src/index.ts

```diff
@@ -651,22 +720,32 @@ export async function startAnalysis<
             if (analysisOptions.platform !== 'android' || analysisOptions.runTarget !== 'emulator')
                 throw new Error('Resetting devices is only supported for Android emulators.');
 
-            const snapshotName = (analysisOptions.targetOptions as RunTargetOptions['android']['emulator'])
-                ?.snapshotName;
+            const snapshotName = emulator?.resetSnapshotName;
             if (!snapshotName) throw new Error('Cannot reset device: No snapshot name specified.');
 
-            return timeout(platform.resetDevice(snapshotName), {
+            return timeout(platform.resetDevice(snapshotName, emulator?.getAbortSignal()), {
                 milliseconds: 5 * 60 * 1000,
+                signal: emulator?.getAbortSignal(),
             }).catch(async (err) => {
                 if (!(err instanceof TimeoutError)) throw err;
 
                 // Sometimes, the Android emulator gets stuck and doesn't accept any commands anymore. In this case, we
                 // restart it.
-                await timeout((await this).ensureDevice({ killExisting: true }), { milliseconds: 60 * 1000 });
-                await timeout(platform.resetDevice(snapshotName), { milliseconds: 5 * 60 * 1000 });
+                // TODO: Maybe we want this to call `emulator.start({forceRestart: true})` directly? We could get rid of another `ensureDevice` this way.
```

Comment:
+1. This way we get rid of the `skipReset` param.

Reply:
Though, given our choice to say that only `ensureDevice()` should deal with emu annoyances, we could also just get rid of this added complexity here. The next run of `ensureDevice()` in the user's code will fix the emu anyway.